src/opm/parser/eclipse/EclipseState/Schedule/VFPProdTable.cpp

```diff
@@ -525,6 +525,35 @@ void VFPProdTable::convertGFRToSI(const GFR_TYPE& type,
 }
 
 
+void VFPProdTable::convertAlqToSI(const ALQ_TYPE& type,
+                                  std::vector<double>& values,
+                                  const UnitSystem& unit_system) {
+    double scaling_factor = 1.0;
+    switch (type) {
+        case ALQ_IGLR:
+            scaling_factor = unit_system.parse("GasSurfaceVolume/LiquidSurfaceVolume/Time").getSIScaling();
+            break;
+        case ALQ_TGLR:
+            scaling_factor = unit_system.parse("GasSurfaceVolume/LiquidSurfaceVolume").getSIScaling();
+            break;
+        case ALQ_GRAT:
+            scaling_factor = unit_system.parse("GasSurfaceVolume/Time").getSIScaling();
+            break;
+        case ALQ_UNDEF:
+            break;
+        case ALQ_PUMP:
+        case ALQ_COMP:
+        case ALQ_INVALID:
+        case ALQ_BEAN:
+            std::runtime_error("scaling of the ALQ type, not implemented ");
```

Comment:
For things like this - which is "bug/limitation" in `flow` I try to use `std::logic_error`.